src/resolution/snapshot.rs

```diff
@@ -268,6 +268,55 @@ impl NpmResolutionSnapshot {
     }
   }
 
+  pub fn subset(&self, package_reqs: &[PackageReq]) -> Option<Self> {
+    if !package_reqs
+      .iter()
+      .all(|req| self.package_reqs.contains_key(req))
+    {
+      return None;
+    }
+
+    let mut new_package_reqs = HashMap::with_capacity(package_reqs.len());
+    let mut packages = HashMap::with_capacity(package_reqs.len() * 2);
+    let mut packages_by_name: HashMap<String, Vec<NpmPackageId>> =
+      HashMap::with_capacity(package_reqs.len());
+    let mut root_packages = HashMap::with_capacity(package_reqs.len());
+
+    let mut visited = HashSet::with_capacity(packages.len());
+
+    let mut stack = Vec::new();
+    for req in package_reqs {
+      let nv = self.package_reqs.get(req)?.clone();
+      let id = self.root_packages.get(&nv)?;
```

Comment:
Instead of using `?` here, I think we should just continue looping (and below whenever this is done).